src/node_nanomsg.cc

```diff
@@ -191,35 +190,40 @@ NAN_METHOD(Err) {
   info.GetReturnValue().Set(Nan::New(nn_strerror(nn_errno())).ToLocalChecked());
 }
 
-typedef struct nanomsg_socket_s {
-  uv_poll_t poll_handle;
-  uv_os_sock_t sockfd;
-  Nan::Callback *callback;
-} nanomsg_socket_t;
+class NanoMsgPollCtx {
+  private:
+  public:
+    uv_poll_t poll_handle;
+    uv_os_sock_t sockfd;
+    Nan::Callback *callback;
+    NanoMsgPollCtx (int s, bool is_sender, Local<v8::Function> cb) {
+      callback = new Nan::Callback(cb);
+      size_t siz = sizeof(uv_os_sock_t);
+      nn_getsockopt(s, NN_SOL_SOCKET, is_sender ? NN_SNDFD : NN_RCVFD, &sockfd,
+          &siz);
+      poll_handle.data = this;
+    }
+    ~NanoMsgPollCtx () {
+      delete callback;
+    }
```

Comment:
No.
However, with C++11, you would get an error with `PollCtx(const int s, const bool is_sender, const Local<v8::Function> cb) : callback(Nan::Callback(cb)) {}` saying `Nan::Callback` is not movable.
With older C++ it would say not copyable.
That is because that line would move (or copy) construct `PollCtx::callback` from the (temporary) `Nan::Callback`.
Without initializer list, it would fail in either version due to `Nan::Callback` not being assignable.